.github/workflows/build.yml

```diff
@@ -0,0 +1,44 @@
+name: release
+
+on:
+  schedule:
+    - cron: '15 3 * * *'  # run daily at 03:15
+
+jobs:
+  rebuild:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: update
+        run: |
+          git submodule status --recursive > before
+          git submodule update --recursive --init
+          git submodule status --recursive > after
+          if diff before after; then
+            echo 'No update required' >&2
+            exit 1
+          fi
+      - name: configure ruby
+        uses: actions/setup-ruby@v1.1.1
+        with:
+          ruby-version: '2.7'
+      - name: install
+        run: |
+          gem install plist
+      - name: build
+        run: |
+          ruby build.rb
+          sed -i "s/Last updated.*:/Last updated: $(date +%Y-%m-%d)/" README.md
+      - name: configure git
+        run: |
+          git config --local 'user.email' 'action@github.com'
+          git config --local 'user.name' 'GitHub Action'
+      - name: commit
+        run: |
+          new_version="$(git -C cylc-textmate-grammar/ rev-parse HEAD)"
+          git commit -a -m "update: ${new_version}"
+      - name: push
+        run: |
+          repo="https://${GITHUB_ACTOR}:${{ secrets.GITHUB_TOKEN }}@github.com/${GITHUB_REPOSITORY}"
```

Comment:
Note I'm not entirely sure what `GITHUB_ACTOR` will be when this action runs on a cron schedule. Simplest solution, try it and find out.